src/components/collaborative-editing/diff/getChangesFromString.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Possible operation types
+ */
+export enum Operation {
+  /**
+   * User inserts symbols
+   */
+  Insert,
+
+  /**
+   * User removes symbols
+   */
+  Remove,
+
+  /**
+   * User replaces symbols
+   */
+  Replace
```

Comment:
It's better to have two operations (Insert & Remove) instead of replace.

Number of transformations is n^2 where n is number of operations. So it's better to keep number of operations as little as possible